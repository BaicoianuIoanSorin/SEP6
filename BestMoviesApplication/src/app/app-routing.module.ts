import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import {OverallInformationComponent} from "./overall-information/overall-information.component";
import {NotFoundComponent} from "src/app/not-found/not-found.component";
import {MoviesComponent} from "src/app/information/movies/movies.component";
import {PeopleComponent} from "src/app/information/people/people.component";

const routes: Routes = [
  {
    path: 'overall-information',
    title: 'Overall Information',
    component: OverallInformationComponent,
  },
  {
    path: 'information',
    children: [
      {
        path:'movies',
        title: 'Information Movies',
        component: MoviesComponent,
      },
      {
        path:'people',
        title: 'Information People',
        component: PeopleComponent,
      }
    ]
  },
  { path: '',
    redirectTo: '/overall-information',
    pathMatch: 'full'
  },
  {
    path: '**',
    title: '404 Page not found',
    component: NotFoundComponent
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
