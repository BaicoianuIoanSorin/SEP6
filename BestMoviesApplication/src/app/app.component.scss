.left-container {
  display: flex;
  align-items: center;
}

.right-container {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-left: auto;
}

.left-container button,
.right-container button {
  margin-right: 10px; /* Adjust the margin as per your preference */
}
