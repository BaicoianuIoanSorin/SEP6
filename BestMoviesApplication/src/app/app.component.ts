import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  template: `
    <bm-layout>
    </bm-layout>
  `,
  styles: [``],
})
export class AppComponent {

}
