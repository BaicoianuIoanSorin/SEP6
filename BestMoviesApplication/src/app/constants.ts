//global constants for the app

import {NbMenuItem} from "@nebular/theme";

//this const will contain all icons used, so another developer doesn't need to search on the internet, if there is something already used

export const PARENT_IDS = {
  OVERALL_INFORMATION_ID: 'overall-information',
  INFORMATION_ID: 'information',
}
export const PAGE_GENERAL_TITLES = {
  //GENERAL MENU
  OVERALL: 'Overall',
  INFORMATION: 'Information'
}

export const PAGE_INFORMATION_TITLES = {
  //INFORMATION MENU
  MOVIES: 'Movies',
  PEOPLE: 'People',
}


export const INFORMATION_MENU_ITEMS: NbMenuItem[] = [
  {
    title: PAGE_INFORMATION_TITLES.MOVIES,
    link: 'information/movies',
    icon: 'film-outline',
    data: {
      id: 'movies'
    }
  },
  {
    title: PAGE_INFORMATION_TITLES.PEOPLE,
    link: 'information/people',
    icon: 'people-outline',
    data: {
      id: 'people'
    }
  }
];

export const GENERAL_MENU_ITEMS: NbMenuItem[] = [
  {
    title: PAGE_GENERAL_TITLES.OVERALL,
    link: '/overall-information',
    icon: 'map-outline',
    data: {
      id: PARENT_IDS.OVERALL_INFORMATION_ID
    }
  },
  {
    title: PAGE_GENERAL_TITLES.INFORMATION,
    icon: 'info-outline',
    children: INFORMATION_MENU_ITEMS,
    data: {
      id: PARENT_IDS.INFORMATION_ID,
    }
  }
]


export function getSelectedItemById(id: string) {
  return GENERAL_MENU_ITEMS.filter(menuItem => {
    if (menuItem.children.length > 0) {
      return menuItem.children.filter(childrenMenuItem => childrenMenuItem.data.id = id);
    }
    return menuItem.data.id = id;
  });
}


//check if parent exists
export function getIndexOfChildInsideParent(parentId: string, childId: string) {
  switch (parentId) {
    case PARENT_IDS.INFORMATION_ID: {
      return INFORMATION_MENU_ITEMS.findIndex(menuItem => menuItem.data.id == childId);
    }
    default: {
      // if parent doesn't exist, then it returns -1
      return -1;
    }
  }
}

export function getIndexOfParent(parentId: string) {
  return GENERAL_MENU_ITEMS.findIndex(menuItem => menuItem.data.id == parentId);
}

export function updateSelectedChildAtIndexOfParent(parentId: string, indexOfChild: number, isSelected: boolean) {
  try {
    switch (parentId) {
      case PARENT_IDS.INFORMATION_ID: {
        INFORMATION_MENU_ITEMS[indexOfChild].selected = isSelected;
        return true;
      }
      default: {
        return false;
      }
    }
  }
  //index out of bounds
  catch (e) {
    console.log(e);
    return false;
  }
}

export function updateSelectedParentAtIndex(indexOfParent: number, isSelected: boolean) {
  try  {
   GENERAL_MENU_ITEMS[indexOfParent].selected = isSelected;
   return true;
  }
  //index out of bounds
  catch (e) {
    console.log(e);
    return false;
  }
}

//pagination
export function paginate(array, page_size, page_number) {
  return array.slice((page_number - 1) * page_size, page_number * page_size);
}

