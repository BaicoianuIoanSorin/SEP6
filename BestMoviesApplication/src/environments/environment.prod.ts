export const environment = {
  production: true,
  test: false,
  //TODO to be added
  api_url: '',
}
