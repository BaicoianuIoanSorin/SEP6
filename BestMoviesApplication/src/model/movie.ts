export class Movie {
  id: number;
  title: string;
  year: number;

  public constructor(id: number, title: string, year: number) {
    this.id = id;
    this.title = title;
    this.year = year;
  }
}
