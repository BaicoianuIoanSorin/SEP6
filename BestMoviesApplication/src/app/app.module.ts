import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import {NbThemeModule, NbLayoutModule, NbActionsModule, NbSidebarModule} from '@nebular/theme';
import { NbEvaIconsModule } from '@nebular/eva-icons';
import { BmLayoutComponent } from '../components/bm-layout/bm-layout.component';

@NgModule({
  declarations: [
    AppComponent,
    BmLayoutComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    BrowserAnimationsModule,
    NbThemeModule.forRoot({name: 'bm-theme'}),
    NbSidebarModule.forRoot(),
    NbLayoutModule,
    NbEvaIconsModule,
    NbActionsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
