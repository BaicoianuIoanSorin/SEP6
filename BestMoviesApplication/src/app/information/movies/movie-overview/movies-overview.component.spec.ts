import {MoviesOverviewComponent} from "./movies-overview.component";
import {createComponentFactory, Spectator, SpectatorFactory} from "@ngneat/spectator/jest";
import {RouterTestingModule} from "@angular/router/testing";
import {BrowserModule} from "@angular/platform-browser";
import {AppRoutingModule} from "../../../app-routing.module";
import {BrowserAnimationsModule} from "@angular/platform-browser/animations";
import {NEBULAR_MODULES, PROVIDERS, STATES} from "../../../app.module";
import {defaultsState} from "./movies-overview.state";
import {NgxsModule, Store} from "@ngxs/store";
import {environment} from "../../../../environments/environment";
import {HttpClientModule} from "@angular/common/http";
import { of } from "rxjs";
import { MovieService } from "../../../../api/movie.service";

const movieServiceMock = {
  getMovieById: jest.fn().mockReturnValue(of([])),
};

describe('app > information > movies > movies-overview > movies-overview.component.spec.ts', () => {
  let store: Store;
  let component: MoviesOverviewComponent;
  let spectator: Spectator<MoviesOverviewComponent>;
  const createComponent: SpectatorFactory<MoviesOverviewComponent> =
    createComponentFactory({
      component: MoviesOverviewComponent,
      imports: [
        RouterTestingModule,
        BrowserModule,
        HttpClientModule,
        AppRoutingModule,
        BrowserAnimationsModule,
        NgxsModule.forRoot(STATES, {
          developmentMode: !environment.production,
        }),
        ...NEBULAR_MODULES,
      ],
      declarations: [],
      providers: [
        ...PROVIDERS,
        Store,
        {
          provide: MovieService,
          useValue: movieServiceMock
        },
      ],
    });

  beforeEach(() => {
    spectator = createComponent({
      detectChanges: false,
    });

    store = spectator.inject(Store);

    store.reset({
      movieOverviewPage: defaultsState,
    })
  });

  it('should create', async () => {
    spectator.detectChanges();
    await spectator.fixture.whenStable();
    spectator.detectChanges();
    await spectator.fixture.whenRenderingDone();

    expect(spectator.component).toBeTruthy();

  });

  it('should match snapshot', async () => {
    spectator.detectChanges();
    await spectator.fixture.whenStable();
    spectator.detectChanges();
    await spectator.fixture.whenRenderingDone();

    // assert
    expect(spectator.fixture).toMatchSnapshot();
  });
});
