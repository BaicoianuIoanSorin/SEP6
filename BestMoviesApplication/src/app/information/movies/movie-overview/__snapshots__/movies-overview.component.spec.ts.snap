// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app > information > movies > movies-overview > movies-overview.component.spec.ts should match snapshot 1`] = `
<app-movies-overview
  TOP_SIZE_LIST={[Function Number]}
  __directors$__selector={[Function Observable]}
  __isFetching$__selector={[Function Observable]}
  __movie$__selector={[Function Observable]}
  __rating$__selector={[Function Observable]}
  __stars$__selector={[Function Observable]}
  __topMovies$__selector={[Function Observable]}
  __topMoviesByRating$__selector={[Function Observable]}
  __topMoviesByYear$__selector={[Function Observable]}
  alive={[Function Boolean]}
  nbToastrService={[Function NbToastrService]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  store={[Function Store]}
>
  <nb-card>
    <nb-card-header>
      <div
        class="person-name"
      >
        <h1>
          <span>
          </span>
        </h1>
        <ul>
          <li>
             Year of publication:  
          </li>
        </ul>
      </div>
      <div
        class="rating"
      >
        <h3>
           No rating for this movie 
        </h3>
      </div>
    </nb-card-header>
    <nb-card-body
      class="grid-container"
    >
      <div
        class="left-content"
      >
        <nb-card
          class="poster-movie-overview size-medium"
          size="medium"
        >
          <nb-card-body
            class="body-poster-movie"
          >
            <img
              alt="null"
              src="../../../assets/no-poster-available.jpg"
            />
            <p>
               Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam eu ullamcorper dui. Ut suscipit urna et neque luctus, in malesuada lacus euismod. Mauris porttitor pellentesque placerat. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Fusce gravida vel lorem vitae mollis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam eu ullamcorper dui. Ut suscipit urna et neque luctus, in malesuada lacus euismod. Mauris porttitor pellentesque placerat. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Fusce gravida vel lorem vitae mollis. 
            </p>
          </nb-card-body>
        </nb-card>
        <div
          class="border"
        />
        <nb-card
          class="movies"
        >
          <nb-card-header>
            <h3>
               Directors and actors 
            </h3>
          </nb-card-header>
          <nb-card-body>
            <ul>
              <li>
                 Lauren Bacall 
              </li>
              <li>
                 Brigitte Bardot 
              </li>
            </ul>
            <ul>
              <li>
                 Lauren Bacall 
              </li>
              <li>
                 Brigitte Bardot 
              </li>
              <li>
                 John Belushi 
              </li>
            </ul>
          </nb-card-body>
        </nb-card>
      </div>
      <nb-card
        class="sidebar"
      >
        <nb-card-header>
          <h5>
             Lists of top 5 movies you might be interested
          </h5>
        </nb-card-header>
        <nb-card-body>
          <h6>
            <nb-icon
              icon="arrow-forward-outline"
              pack="eva"
            >
              <svg
                class="eva eva-arrow-forward-outline"
                fill="currentColor"
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 2"
                >
                  <g
                    data-name="arrow-forward"
                  >
                    <rect
                      height="24"
                      opacity="0"
                      transform="rotate(-90 12 12)"
                      width="24"
                    />
                    <path
                      d="M5 13h11.86l-3.63 4.36a1 1 0 0 0 1.54 1.28l5-6a1.19 1.19 0 0 0 .09-.15c0-.05.05-.08.07-.13A1 1 0 0 0 20 12a1 1 0 0 0-.07-.36c0-.05-.05-.08-.07-.13a1.19 1.19 0 0 0-.09-.15l-5-6A1 1 0 0 0 14 5a1 1 0 0 0-.64.23 1 1 0 0 0-.13 1.41L16.86 11H5a1 1 0 0 0 0 2z"
                    />
                  </g>
                </g>
              </svg>
            </nb-icon>
             Top movies of all time 
          </h6>
          <nb-list
            role="list"
          >
          </nb-list>
          <nb-list
            role="list"
          >
          </nb-list>
          <nb-list
            role="list"
          >
          </nb-list>
        </nb-card-body>
      </nb-card>
    </nb-card-body>
  </nb-card>
</app-movies-overview>
`;
